install.sh

```diff
@@ -524,6 +530,59 @@ __package_installed() {
 	fi
 }
 
+__package_version() {
+	if ! __package_installed "$1"; then
+		echo ""
+	fi
+
+	if [ $_PKG_MGR -eq 1 ]; then # apt
+		echo $(dpkg-query -W -f '${Version}' "$1")
+	elif [ $_PKG_MGR -eq 2 ]; then # yum and dnf
+		echo $(rpm -qa --queryformat='%{VERSION}' "$1")
+	fi
+}
+
+__satisfies_version() {
+	local installed="$1"
+	local desired="$2"
+	local installed_major="$(echo $installed | cut -d'.' -f1)"
+	local installed_minor="$(echo $installed | cut -d'.' -f2)"
+	local installed_patch="$(echo $installed | cut -d'.' -f3)"
+	local desired_major="$(echo $desired | cut -d'.' -f1)"
+	local desired_minor="$(echo $desired | cut -d'.' -f2)"
+	local desired_patch="$(echo $desired | cut -d'.' -f3)"
+
+	# Set any empty values to 0. echo > /dev/null is to prevent invoking values as commands
+	# http://wiki.bash-hackers.org/syntax/pe#assign_a_default_value
+	echo ${installed_major:=0} > /dev/null
+	echo ${installed_minor:=0} > /dev/null
+	echo ${installed_patch:=0} > /dev/null
+	echo ${desired_major:=0} > /dev/null
+	echo ${desired_minor:=0} > /dev/null
+	echo ${desired_patch:=0} > /dev/null
+
+	if [ "$installed_major" -lt "$desired_major" ]; then
+		false; return
```

Comment:
With bash functions a return value of 1 represents false and 0 represents true since this is convention for program return codes and what `test` expects. `return 1` isn't really intuitive coming from other languages and something like this just seems like a cruel joke:
```
local true=0
local false=1
...
return $true
```
So `false; return` seemed the like the least confusing option. It could also be written on two lines but I figured it was close enough to `return false;` to be ok as-is.